src/state/actions/downloads.ts

```diff
@@ -47,19 +47,23 @@ export const initDownloads = async () => {
   const downloadedPodcastEpisodeCounts = await getDownloadedPodcastEpisodeCountsService()
   const downloadedPodcasts = await getDownloadedPodcastsService()
   const autoDownloadSettings = await getAutoDownloadSettingsService()
-  const downloadedEpisodeLimitCount = await AsyncStorage.getItem(PV.Keys.DOWNLOADED_EPISODE_LIMIT_GLOBAL_COUNT)
-  const downloadedEpisodeLimitDefault = await AsyncStorage.getItem(PV.Keys.DOWNLOADED_EPISODE_LIMIT_GLOBAL_DEFAULT)
+  const downloadedEpisodeLimitCount = AsyncStorage.getItem(PV.Keys.DOWNLOADED_EPISODE_LIMIT_GLOBAL_COUNT)
+  const downloadedEpisodeLimitDefault = AsyncStorage.getItem(PV.Keys.DOWNLOADED_EPISODE_LIMIT_GLOBAL_DEFAULT)
 
-  setGlobal({
-    autoDownloadSettings,
-    downloadsActive,
-    downloadsArray,
-    downloadedEpisodeIds,
-    downloadedEpisodeLimitCount,
-    downloadedEpisodeLimitDefault,
-    downloadedPodcastEpisodeCounts,
-    downloadedPodcasts
-  })
+  // TODO: There is a race condition preventing this state from being set properly on app launch :(
+  // I don't know where the problem is coming from...
```

Comment:
setGlobal is Asynchronous by nature. What values are you looking for and where that you expected them to be populated in time?

Reply:
Something weird was happening where the state would load correctly on the home screen, with all the subscribed podcasts, but then quickly disappear and be replaced with an empty global state. initDownloads is only called in one place on app launch. Maybe there is a timing issue with getAuthUserInfo or initPlayerState initializeScreenData?

Reply:
@kreonjr when I remove the setTimeout, I am unable to reproduce locally...I spent a few hours trying to fix this issue, and I was having some nasty cache issues that I couldn't seem to fix. Anyway, I'm not sure this race condition even actually exists. I'd say it doesn't hurt to use a setTimeout in initDownloads though so I'm inclined to leave it here.